generate_chip.py

```diff
@@ -0,0 +1,345 @@
+                max_y=max_y,
+                excess_x=courtyard_excess,
+                excess_y=courtyard_excess,
+            )))
+
+            # Labels
+            if config.width < 2.0:
+                offset = label_offset_thin
+            else:
+                offset = label_offset
+            dy = ff(config.width / 2 + offset)  # y offset (delta-y)
+            text_attrs = '(height {}) (stroke_width 0.2) ' \
+                         '(letter_spacing auto) (line_spacing auto)'.format(pkg_text_height)
+            lines.append('  (stroke_text {} (layer top_names)'.format(uuid_text_name))
+            lines.append('   {}'.format(text_attrs))
+            lines.append('   (align center bottom) (position 0.0 {}) (rotation 0.0)'.format(dy))
+            lines.append('   (auto_rotate true) (mirror false) (value "{{NAME}}")')
+            lines.append('  )')
+            lines.append('  (stroke_text {} (layer top_values)'.format(uuid_text_value))
+            lines.append('   {}'.format(text_attrs))
+            lines.append('   (align center top) (position 0.0 -{}) (rotation 0.0)'.format(dy))
+            lines.append('   (auto_rotate true) (mirror false) (value "{{VALUE}}")')
+            lines.append('  )')
+
+            lines.append(' )')
+
+        add_footprint_variant('density~b', 'Density Level B (median protrusion)', 'B', 0.0)
+        add_footprint_variant('density~a', 'Density Level A (max protrusion)', 'A', 0.0)
+        # add_footprint_variant('density~hs', 'Hand Soldering', 'A', handsoldering_toe_extension)
+
+        lines.append(')')
+
+        pkg_dir_path = path.join(dirpath, uuid_pkg)
+        if not (path.exists(pkg_dir_path) and path.isdir(pkg_dir_path)):
+            makedirs(pkg_dir_path)
+        with open(path.join(pkg_dir_path, '.librepcb-pkg'), 'w') as f:
+            f.write('0.1\n')
+        with open(path.join(pkg_dir_path, 'package.lp'), 'w') as f:
+            f.write('\n'.join(lines))
+            f.write('\n')
+
+
+if __name__ == '__main__':
+    def _make(dirpath: str):
+        if not (path.exists(dirpath) and path.isdir(dirpath)):
+            makedirs(dirpath)
+    _make('out')
+    _make('out/chip')
+    _make('out/chip/pkg')
+    # Chip resistors (RESC)
+    generate_pkg(
+        dirpath='out/chip/pkg',
+        author='Danilo B.',
+        name='RESC{size_metric} ({size_imperial})',
+        description='Generic chip resistor {size_metric} (imperial {size_imperial}).\\n\\n'
+                    'Length: {length}mm\\nWidth: {width}mm',
+        configs=[
+            # Configuration: Values taken from Samsung specs.
+            #        imperial, len, wid,  hght, gap
+            ChipConfig('01005', .4,  .2,  0.15, 0.2),   # noqa
+            ChipConfig('0201',  .6,  .3,  0.26, 0.28),  # noqa
+            ChipConfig('0402', 1.0,  .5,  0.35, 0.5),   # noqa
+            ChipConfig('0603', 1.6,  .8,  0.55, 0.8),   # noqa
+            ChipConfig('0805', 2.0, 1.25, 0.70, 1.4),   # noqa
```

Comment:
The gap seems to be too high here. Parts of the leads would be placed on the soldermask with such a large gap. The Samsung datasheet is very strange regarding the 2012 footprint, Vishay and resistorguide.com recommend a gap of 1.2mm which makes much more sense to me.

```suggestion
            ChipConfig('0805', 2.0, 1.25, 0.70, 1.2),   # noqa
```